modelservice/worker.py

```diff
@@ -72,12 +78,118 @@ def register(self):
         info(f"Received request to build models: {request}")
-        return BuildModelsResponse(
+
+        worker_timer: Timer = Timer()
+        worker_timer.start()
+
+        feature_fields: list = list(request.feature_fields)
+        label_field: str = request.label_field
+        hyper_parameters: HyperParameters = request.hyper_parameters
+        epochs: int = hyper_parameters.epochs
+        learning_rate: float = hyper_parameters.learning_rate
+        normalize_inputs: bool = hyper_parameters.normalize_inputs
+        train_split: float = hyper_parameters.train_split
+        test_split: float = 1.0 - train_split
+
+        if hyper_parameters.optimizer_type == OptimizerType.ADAM:
+            optimizer = tf.keras.optimizers.Adam(learning_rate)
+        else:
+            optimizer = tf.keras.optimizers.SGD(learning_rate)
+
+        if hyper_parameters.loss_type == LossType.MEAN_SQUARED_ERROR:
+            loss = "mean_squared_error"
+        elif hyper_parameters.loss_type == LossType.MEAN_SQUARED_ERROR:
+            loss = "mean_squared_error"
+        else:
+            loss = "mean_absolute_error"
+
+        evaluation_metrics: list = []  # list(EvaluationMetric)
+
+        # Make models dir for job id
+        models_dir: str = f"{self.data_dir}/{request.id}"
+        os.mkdir(models_dir)
+
+        count = 1
+        gis_join_timer: Timer = Timer()
+        for gis_join in request.gis_joins:
+
+            gis_join_timer.start()
+            info(f"Loading data for GISJOIN {gis_join} ({count}/{len(request.gis_joins)})...")
+
+            # Load data
+            csv_path: str = f"{self.data_dir}/{gis_join}.csv"
+            all_df: pd.DataFrame = pd.read_csv(csv_path, header=0).drop("GISJOIN", 1)
+            len_df: int = len(all_df.index)
+            info(f"Loaded data for GISJOIN {gis_join}: {len_df} records")
+            if normalize_inputs:
+                scaled = MinMaxScaler(feature_range=(0, 1)).fit_transform(all_df)
+                all_df = pd.DataFrame(scaled, columns=all_df.columns)
+
+            features = all_df[feature_fields]
+            labels = all_df[label_field]
+
+            # Create Sequential model
+            model = tf.keras.Sequential()
+
+            # Add input layer
+            model.add(tf.keras.Input(shape=(len(request.feature_fields))))
+
+            # Add hidden layers
+            for hidden_layer in hyper_parameters.hidden_layers:
+                name: str = hidden_layer.name
+                units: int = hidden_layer.units
+                activation = "relu"
+                model.add(tf.keras.layers.Dense(units=units, activation=activation, name=name))
+
+            # Add output layer
+            output_layer: OutputLayer = hyper_parameters.output_layer
+            name: str = output_layer.name
+            activation = "relu"
+            model.add(tf.keras.layers.Dense(units=1, activation=activation, name=name))
```

Comment:
👍  for adding hidden and output layer data in the hyperparameters and for keeping it simple and just using relu.